invokeai/app/util/torch_cuda_allocator.py

```
@@ -0,0 +1,42 @@
+import logging
+import os
+
+
+def configure_torch_cuda_allocator(pytorch_cuda_alloc_conf: str, logger: logging.Logger | None = None):
+    """Configure the PyTorch CUDA memory allocator. See
+    https://pytorch.org/docs/stable/notes/cuda.html#optimizing-memory-usage-with-pytorch-cuda-alloc-conf for supported
+    configurations.
+    """
+
+    # Raise if the PYTORCH_CUDA_ALLOC_CONF environment variable is already set.
+    prev_cuda_alloc_conf = os.environ.get("PYTORCH_CUDA_ALLOC_CONF", None)
+    if prev_cuda_alloc_conf is not None:
+        raise RuntimeError(
+            f"Attempted to configure the PyTorch CUDA memory allocator, but PYTORCH_CUDA_ALLOC_CONF is already set to "
+            f"'{prev_cuda_alloc_conf}'."
+        )
```

It looks like there are 3 options:
- Raise if env var is set and Invoke setting is set
- Do nothing, letting env var manage the setting
- Override the env var w/ the Invoke setting

Is that correct? Why choose the raise option?

---

Correct. If `PYTORCH_CUDA_ALLOC_CONF` is set, this means that the user has manually configured the value in two places (the env var and the invoke config). I figured the best thing to do was to fail loudly rather than quietly choose one.